crates/snforge-scarb-plugin/src/parse.rs

```diff
@@ -0,0 +1,65 @@
+use crate::attributes::{AttributeInfo, ErrorExt};
+use cairo_lang_diagnostics::DiagnosticsBuilder;
+use cairo_lang_filesystem::ids::{FileKind, FileLongId, VirtualFile};
+use cairo_lang_macro::Diagnostic;
+use cairo_lang_parser::{parser::Parser, utils::SimpleParserDatabase};
+use cairo_lang_syntax::node::{
+    ast::{FunctionWithBody, ModuleItem, OptionArgListParenthesized},
+    db::SyntaxGroup,
+    helpers::QueryAttrs,
+};
+use cairo_lang_utils::Upcast;
+use indoc::formatdoc;
+use std::sync::Arc;
+
+pub fn parse<T: AttributeInfo>(
+    code: &str,
+) -> Result<(SimpleParserDatabase, FunctionWithBody), Diagnostic> {
+    let simple_db = SimpleParserDatabase::default();
+    let code = Arc::new(code.to_string());
+    let db: &dyn SyntaxGroup = simple_db.upcast();
+    let virtual_file = db.intern_file(FileLongId::Virtual(VirtualFile {
```

Comment:
// TODO(#2357): Use `db.parse_virtual` here instead of creating the virtual file manually
```suggestion
    let virtual_file = db.intern_file(FileLongId::Virtual(VirtualFile {
```